cranelift-codegen/src/verifier/flags.rs

```diff
@@ -50,12 +50,12 @@ impl<'a> FlagsVerifier<'a> {
     fn check(&mut self, errors: &mut VerifierErrors) -> VerifierStepResult<()> {
         // List of EBBs that need to be processed. EBBs may be re-added to this list when we detect
         // that one of their successor blocks needs a live-in flags value.
-        let mut worklist = SparseSet::new();
+        let mut worklist = EntitySet::new();
         for ebb in self.func.layout.ebbs() {
             worklist.insert(ebb);
         }
 
-        while let Some(ebb) = worklist.pop() {
+        for ebb in worklist.keys() {
```

Comment:
This changes behavior. Before only the ebbs which were in func.layout or were inserted into the worklist at https://github.com/XAMPPRocky/cranelift/blob/f3cb557807e2fa931c235172bba17dd8e21fa613/cranelift-codegen/src/verifier/flags.rs#L66 were visited. Now all ebbs in the range ebb0..the ebb in func.layout with the highest ebb index is visited. This causes extra ebbs to be visited, and it misses the ebbs inserted at https://github.com/XAMPPRocky/cranelift/blob/f3cb557807e2fa931c235172bba17dd8e21fa613/cranelift-codegen/src/verifier/flags.rs#L66 with index higher than the highest in func.layout to be skipped.